include/ignition/gazebo/Util.hh

```diff
@@ -51,6 +51,28 @@ namespace ignition
       const EntityComponentManager &_ecm, const std::string &_delim = "/",
       bool _includePrefix = true);
 
+    /// \brief Helper function to get an entity given its scoped name.
+    /// The scope may start at any level. For example, in this hierarchy:
+    ///
+    /// world_name
+    ///  model_name
+    ///    link_name
+    ///
+    /// All these names will return the link entity:
+    ///
+    /// * world_name::model_name::link_name
+    /// * model_name::link_name
+    /// * link_name
+    ///
+    /// \param[in] _scopedName Entity's scoped name.
+    /// \param[in] _ecm Immutable reference to ECM.
+    /// \param[in] _delim Delimiter between names, defaults to "::".
+    /// \return The engity or kNullEntity if no entity was found with that
+    /// name.
+    Entity IGNITION_GAZEBO_VISIBLE entityFromScopedName(
+      const std::string &_scopedName, const EntityComponentManager &_ecm,
```

Comment:
Ah good point, I was living in a world with unique entity names :woman_facepalming: 

Requiring an entity may work well for server plugins that are attached to entities, but GUI plugins don't really have a reference entity, so it would be cumbersome to figure out what entity we're `relative_to`. And I don't like the idea of requiring fully scoped names mainly because it makes it difficult to use the same GUI config with various worlds.

So, I have an alternative suggestion in a1cf2841ea2c57f3ef1e4bad464e76cfe829916b, which is to return all entities that match the scoped name. Then the caller can decide what to do with all of them. Let me know whether that works for your use case. I may have pushed the cumbersomeness to your side :sweat_smile:

Reply:
It does indeed become very cumbersome for system writers if they have to sift through a list of entities to determine which entity they should pick. If the issue is figuring out what the world name should be, I think that information is available in `GuiRunner`, so it shouldn't be too difficult to pass it on to `GuiSystem`s. Then, we can have scoped names in GUI configs without needing the world name

Reply:
Fair enough. Added `relativeTo` in 673388d6d058adf8c4dd4d4bc0d89d65c7c82053.

I kept the logic returning multiple entities just in case. Gazebo isn't enforcing unique names, so it isn't an impossibility. I can also imagine this being used to return multiple entities on purpose, for example:

```
model1
  the_link
    visual
model2
  the_link
    visual
```

If I want all visuals from all `the_link`s, I can get them with `entitiesFromScopedName("the_link::visual", ecm)`

Reply:
@azeey , I'll wait on your ok on this function before merging

Reply:
Looks good. Thanks!